### ISSUE TITLE
[Feature] Migrate workflows to reference shared actions

### ISSUE THREAD

**Issue by mikealfare** [dbt-labs/dbt-postgres#74]
### Is this your first time submitting a feature request?

- [X] I have read the [expectations for open source contributors](https://docs.getdbt.com/docs/contributing/oss-expectations)
- [X] I have searched the existing issues, and I could not find an existing issue for this feature
- [X] I am requesting a straightforward extension of existing dbt-postgres functionality, rather than a Big Idea better suited to a discussion

### Describe the feature

https://github.com/dbt-labs/actions/issues/173 migrates actions from `dbt-labs/dbt-adapters` into `dbt-labs/actions`. `dbt-postgres` should start using the new actions remove its dependency from `dbt-labs/dbt-adapters` in this area so that `dbt-labs/dbt-adapters` can retire these actions.

**Comment by leahwicz** [comment]
@mikealfare would you want any help with this? Feel free to tell me to buzz off 😄 but I had the itch to do some DevOps work lol

**Comment by mikealfare** [comment]
> @mikealfare would you want any help with this? Feel free to tell me to buzz off 😄 but I had the itch to do some DevOps work lol

@leahwicz The three stories attached above are ready to go. If you're willing to work them we would definitely appreciate it! Let me know if you have questions, and bare in mind that we're at our team retreat next week so I'll be less responsive during that period.

**Comment by leahwicz** [comment]
@mikealfare yup I'll take a crack at #137 to start and you can let me know what you think

**Comment by mikealfare** [comment]
This is rolling up into the monorepo work. I removed the one story that has incremental work on the monorepo so that I can close this one as a duplicate.
